Add config prop

nathanieljramirez opened the pull request:
You can add config for support to customize the default config of a ruffle player. If you would like to see an example of one using the official docs I have simplified it. [Ruffle config](https://ruffle.rs/js-docs/master/interfaces/BaseLoadOptions.html)

nathanieljramirez committed:
add config support

nathanieljramirez commented:
This works so you can disable preloader, context menu or turn on autoplay.

lacymorrow commented:
I like it. I'll merge it later tonight but I think we should add types to match the ruffle doc options. 

I'll have copilot generate them if you like.

nathanielrz commented:
> I like it. I'll merge it later tonight but I think we should add types to match the ruffle doc options. 
> 
> I'll have copilot generate them if you like. 

Sounds good

nathanieljramirez commented:
Sounds great

nathanieljramirez committed:
add types to every config

nathanieljramirez commented:
I just added to types for you it took like 2 hours but it was definitely worth it since this project is absolutely incredible. You can review the code if you would like, I also added the default config used by ruffle.

lacymorrow reviewed (approved):
LGTM++

lacymorrow commented:
Thanks for adding types, I'm going to tweak slightly to move them into a new file.

lacymorrow commented:
Ahhh sorry it took you so long, you're going to hate it, but Copilot generates them immediately.

If you don't have Copilot access, I've been test driving Cursor IDE, which is a drop-in replacement for VSCode with their own CursorAI built in (Copilot Competitor)

nathanieljramirez commented:
> Ahhh sorry it took you so long, you're going to hate it, but Copilot generates them immediately.
> 
> 
> 
> If you don't have Copilot access, I've been test driving Cursor IDE, which is a drop-in replacement for VSCode with their own CursorAI built in (Copilot Competitor)

I should try it isn't it the extension where you just have to press tab?

lacymorrow commented:
> I should try it isn't it the extension where you just have to press tab?

That's the one. They aren't perfect yet but at this point I rely on copilot to generate the tedious parts of the code.
I have pretty bad RSI from using my keyboard, generating code saves my hands!

lacymorrow commented:
I've published v1.0.3 on [NPM](https://www.npmjs.com/package/react-ruffle)
